# This config file is used by ros2_control
controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController


    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

joint_trajectory_controller:
  ros__parameters:
    joints:
      - link1_revolve
      - link2_revolve
      - base_revolve
      - link3_revolve
      - link4_revolve
    command_interfaces:
      - position
      - velocity
    state_interfaces:
      - position
      - velocity
    allow_partial_joints_goal: false
    allow_integration_in_goal_trajectories: true
    allow_nonzero_velocity_at_trajectory_end: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 5.0
      goal_tolerance: 0.001
      link1_revolve: {trajectory: 0.2, goal: 0.1}
      link2_revolve: {trajectory: 0.2, goal: 0.1}
      base_revolve: {trajectory: 0.2, goal: 0.1}
      link3_revolve: {trajectory: 0.2, goal: 0.1}
      link4_revolve: {trajectory: 0.2, goal: 0.1}